### ISSUE TITLE
Hoppers picking up dropped items even when a chest is above the hopper.

### ISSUE THREAD

**Issue by NullCase** [PaperMC/Paper#993]
# Performance issue
Hopper bug, picking up items even when blocked by chest.

###  Timings or Profile link:
https://timings.aikar.co/?id=00b83919c87d4a588451a969754e6e9e

### Description of issue:
Hoppers will take up dropped items even if a chest is over the hopper/would ordinarily block the hopper from picking up items.

### bukkit.yml, spigot.yml, paper.yml, server.properties
___Gist/pastebin/hastebin links___

### Paper build number:
git-Paper-1313 for MC 1.12.2

# Incompatibility (Bug)

### What behaviour is expected:
Chest over hopper should block hopper from picking up dropped items.

### What behaviour is observed:
The opposite.

### Steps/models to reproduce:
Put chest over hopper. throw item "into the hopper".

**Comment by zachbr** [comment]
Should be fixed in builds 1317+

**Comment by NullCase** [comment]
thanks!

**Comment by NullCase** [comment]
I have tested 1317 now. Not fixed in this version.

1318+?

Thanks for the quick response.

**Comment by NullCase** [comment]
Tested 1317 today. Not solved yet.

When Hoppers: Push-based = true

Then dropped items enter blocked hoppers (chest over hopper).

**Comment by electronicboy** [comment]
Is this only an issue with push-based hoppers?

**Comment by aikar** [comment]
can you try https://github.com/PaperMC/Paper/pull/976#issuecomment-360342313 ?

I made some changes / improvements around there.

**Comment by NullCase** [comment]
>Is this only an issue with push-based hoppers?

Seems so.

**Comment by NullCase** [comment]
>can you try #976 (comment) ?

@aikar Did you want me to test that? Unclear who you meant.

**Comment by zachbr** [comment]
Push based hopper was dropped, this ticket is no longer relevant.
